src/app/pages/courses-page/courses-implementation.service.ts

```diff
@@ -11,39 +13,33 @@ const BACKEND_URL = 'http://localhost:3004/';
   providedIn: 'root',
 })
 export class CoursesImplementationService  implements CoursesService {
-  private _courses = new BehaviorSubject<Course[]>([]);
   private _loading = new BehaviorSubject<boolean>(false);
-  private dataStore: { courses: Course[] } = { courses: [] };
 
   start = 0;
   count = 5;
   sort = 'date';
   textFragment = '';
 
-  constructor(private http: HttpClient) { }
+  constructor(private http: HttpClient, private store: Store) { }
 
-  get courses(): Observable<Course[]> {
-    return this._courses.asObservable();
-  }
 
   get loading(): Observable<boolean> {
     return this._loading.asObservable();
   }
 
   loadAll(): void {
     this._loading.next(true);
+    this.start = 0;
 
     this.getCourses(this.start, this.count, this.sort, this.textFragment).subscribe(
```

Comment:
I think we can remove service and use effects of ngrx. All logic with loading/creating and communication with API should be moved in effects.